server/src/main/java/io/crate/planner/operators/JoinPlanBuilder.java

```diff
@@ -192,13 +192,29 @@ private static LogicalPlan joinWithNext(Function<AnalyzedRelation, LogicalPlan>
 
         JoinPair joinPair = removeMatch(joinPairs, joinNames, nextName);
         final JoinType type;
-        final Symbol condition;
+        ArrayList<Symbol> conditions = new ArrayList<>();
         if (joinPair == null) {
             type = JoinType.CROSS;
-            condition = null;
         } else {
             type = maybeInvertPair(nextName, joinPair);
-            condition = joinPair.condition();
+            if (joinPair.condition() != null) {
+                conditions.add(joinPair.condition());
+            }
+            // There could be additional joinPairs that are not directly connected to the current joinPair,
+            // when there are WHERE clauses referencing to multiple relations of a previous join which was converted
+            // into a joinPair by JoinOperations.convertImplicitJoinConditionsToJoinPairs(...)
+            // Example:
+            //  t1. JOIN t2 ON t1.x = t2.y JOIN t3 ON t1.x = t3.z WHERE t2.y = t3.z
+            //  -> JoinPair(t1, t2)     <- 1st JOIN
```

Comment:
Just curious what happens if we try to add same pair implicitly and explicitly:
```
SELECT * FROM j1 
  JOIN j2 ON j1.x = j2.y 
  JOIN j3 ON j1.x = j2.y AND j2.y = j3.z 
WHERE j2.y = j3.z
```
I guess (j2,j3) will be handled twice? If if it's the case, don't think it's a problem(?), we would just check same condition twice.

Reply:
@BaurzhanSakhariev 
Good point, but this isn't inside the scope of this PR as this happens already if the same explicit condition is added twice.

```
cr> EXPLAIN SELECT * FROM j1  JOIN j2 ON j1.x = j2.y AND j1.x = j2.y 
+----------------------------------+
| EXPLAIN                          |
+----------------------------------+
| HashJoin[((x = y) AND (x = y))]  |
|   ├ Collect[doc.j1 | [x] | true] |
|   └ Collect[doc.j2 | [y] | true] |
+----------------------------------+
```


And yes, performance wise we should detect any duplication and remove it to avoid useless evaluation on the same function. I've created an issue for this #13810, but I'm not sure how important this is/how big the performance impact is.

Reply:
UPD: Same pair case can be even trickier, 
implicit can be `ON ... AND j2.y = j3.z ` and implicit same thing but swapped: `WHERE j3.z = j2.y`

I guess we will have 2 pairs (j2,j3) (j3,j2) - but again I don't think it's a big problem

Reply:
>how big the performance impact is.

I agree, don't think it's a big deal